src/apps/frontend/services/task.service.ts

```diff
@@ -0,0 +1,42 @@
+import { AccessToken, ApiResponse } from '../types';
+import { Task } from '../types/task';
+
+import APIService from './api.service';
+
+export default class TaskService extends APIService {
+  getTasks = async (): Promise<ApiResponse<Task[]>> => {
+    const userAccessToken = JSON.parse(
+      localStorage.getItem('access-token'),
+    ) as AccessToken;
+    return this.apiClient.get('/tasks', {
+      headers: {
+        Authorization: `Bearer ${userAccessToken.token}`,
+      },
+    });
+  };
+
+  updateTask = async (
+    taskId: string,
+    taskData: Partial<Task>,
+  ): Promise<ApiResponse<Task>> => {
+    const userAccessToken = JSON.parse(
+      localStorage.getItem('access-token'),
+    ) as AccessToken;
+    return this.apiClient.put(`/tasks/${taskId}`, taskData, {
```

Comment:
I think the concept of JSON object ans JS object is still not clear.

Server returns a JSON , say `{date: '2020-12-20'}`. The job of the service is to convert that into Javascript object `new Task(json)`. This is true in `getTasks`